dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/api/RequestApi.kt

```diff
@@ -91,5 +95,101 @@ interface RequestApi {
+
+    /**
+     * A method to post a single request to Dataland.
+     * @param singleDataRequest includes necessary info for the single request
+     * @return response after posting a single data request to Dataland
+     *
+     */
+    @Operation(
+        summary = "Send a single request",
+        description = "A single of data requests for specific frameworks and companies is being sent.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully processed a single data request."),
+        ],
+    )
+    @PostMapping(
+        value = ["/single"],
+        consumes = ["application/json"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun postSingleDataRequest(
+        @Valid @RequestBody
+        singleDataRequest: SingleDataRequest,
+    ):
+        ResponseEntity<List<StoredDataRequest>>
+
+    /** A method for users to get a data request by its ID.
+     * @return the data requests corresponding to the provided ID
+     */
+    @Operation(
+        summary = "Get all stored data requests of the user making the request.",
+        description = "Gets all the stored data request created by the user who is making the request.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data request for the provided ID."),
+        ],
+    )
+    @GetMapping(
+        value = ["/{dataRequestId}"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun getDataRequestById(@PathVariable dataRequestId: UUID): ResponseEntity<StoredDataRequest>
+
+    /** Changes request status of existing data request
+     * @return the modified data request
+     */
+
+    @Operation(
+        summary = "Update status of data request.",
+        description = "Updates status of data request given data request id.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully patched data request."),
+        ],
+    )
+    @PatchMapping(
+        value = ["/{dataRequestId}/requestStatus"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun patchDataRequest(
+        @PathVariable dataRequestId: UUID,
+        @RequestParam requestStatus: RequestStatus = RequestStatus.Open,
+    ): ResponseEntity<StoredDataRequest>
+
+    /** A method for searching data requests based on filters.
+     * @return all data requests of the user in a list
+     */
+    @Operation(
+        summary = "Get all stored data requests based on filters.",
+        description = "Gets all the stored data request based on filters.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data requests"),
+        ],
+    )
+    @GetMapping(
+        value = ["/all"],
```

Comment:
Couldn't this also be directly behind /requests?
So basicaly a "GET" request to the /requests endpoint would return all the request. Or do we really need this addtional /all here?

Reply:
It is used to differentiate the GET endpoints as we have four of them starting with /requests

Reply:
I would suggest to keep the paths "/requests/user" and "/requests/{dataRequestId}" as they fit the respective logic behind (retrieve requests for the user that is currently logged in & get a request by its ID). However, "/requests" would be better in my opinion for searching based on various filters (this would be in line with how we do handle that for companies) and we could then use "/requests/aggregated" for the aggregated endpoint to reflect that it queries requests a bit differently

Reply:
i like your suggestion @PaulChristiansen

Reply:
Good, then I will change the paths as described

Reply:
This is done